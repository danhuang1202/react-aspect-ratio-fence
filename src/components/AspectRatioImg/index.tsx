import React, { FunctionComponent, ReactNode } from 'react'
import AspectRatioFence from '../AspectRatioFence'
import AspectRatioFenceLegacy from '../AspectRatioFenceLegacy'

// @ts-ignore
import styles from './style.css'

type Props = {
  /** Custom class name */
  customClass?: string
  /** The aspect ratio of an image describes the proportional relationship between width and height. */
  ratio: number
  /** The src attribute of <img /> tag */
  src: string
  /** Attributes of <img /> tag */
  attributes?: object
  /**
   *  is only support modern browser.
   *  AspectRatioFence use css new feature of custom property(css variable),
   *  which does not support in IE. Ths fallback solution is AspectRatioFenceLegacy
   *
   *  @default true
   **/
  isModern?: boolean
  /** custom children node */
  children?: ReactNode
}

const AspectRatioFenceImg: FunctionComponent<Props> = ({
  ratio,
  src,
  attributes,
  customClass,
  isModern = true,
  children
}) => {
  const Container = isModern ? AspectRatioFence : AspectRatioFenceLegacy
  return (
    <Container customClass={customClass} ratio={ratio}>
      <img alt="" src={src} className={styles.img} {...attributes} />
      {children}
    </Container>
  )
}

export default AspectRatioFenceImg
