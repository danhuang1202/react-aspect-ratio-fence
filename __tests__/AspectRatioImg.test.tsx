import React from 'react'
import renderer from 'react-test-renderer'
import AspectRatioImg from '@components/AspectRatioImg'

describe('[snapshot] AspectRatioImg', () => {
  const defaultProps = {
    ratio: 16 / 9,
    src: 'aspect-ratio-640w.jpg',
    attributes: {
      alt: 'AspectRatioImg',
      srcSet: `
        aspect-ratio-320w.jpg,
        aspect-ratio-480w.jpg 1.5x,
        aspect-ratio-640w.jpg 2x`
    }
  }

  it('AspectRatioImg in modern browser', () => {
    const dom = renderer.create(<AspectRatioImg {...defaultProps} />)
    expect(dom).toMatchSnapshot()
  })

  it('AspectRatioImg in legacy browser', () => {
    const props = {
      ...defaultProps,
      isModern: false
    }

    const dom = renderer.create(<AspectRatioImg {...props} />)
    expect(dom).toMatchSnapshot()
  })
})
