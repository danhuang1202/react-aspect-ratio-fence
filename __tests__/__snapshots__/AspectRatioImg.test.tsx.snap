// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[snapshot] AspectRatioImg AspectRatioImg in legacy browser 1`] = `
<div
  className="AspectRatioFenceLegacy_wrap_ZYFye"
>
  <div
    className="AspectRatioFenceLegacy_container_3VjLm"
    style={
      Object {
        "paddingBottom": "56.25%",
      }
    }
  >
    <img
      alt="AspectRatioImg"
      className="AspectRatioImg_img_-WkaR"
      src="aspect-ratio-640w.jpg"
      srcSet="
        aspect-ratio-320w.jpg,
        aspect-ratio-480w.jpg 1.5x,
        aspect-ratio-640w.jpg 2x"
    />
  </div>
</div>
`;

exports[`[snapshot] AspectRatioImg AspectRatioImg in modern browser 1`] = `
<div
  className="AspectRatioFence_wrap_2HH2L"
  ratio={1.7777777777777777}
  style={
    Object {
      "--ratio": "(1.7777777777777777)",
    }
  }
>
  <img
    alt="AspectRatioImg"
    className="AspectRatioImg_img_-WkaR"
    src="aspect-ratio-640w.jpg"
    srcSet="
        aspect-ratio-320w.jpg,
        aspect-ratio-480w.jpg 1.5x,
        aspect-ratio-640w.jpg 2x"
  />
</div>
`;
